tools/ci_connector_ops/ci_connector_ops/qa_engine/cloud_availability_updater.py

```diff
@@ -50,31 +61,64 @@ def update_definitions_mask(connector: ConnectorQAReport, definitions_mask_path:
 
     with open(definitions_mask_path, "a") as f:
         f.write(to_append)
-    logging.info(f"Updated {definitions_mask_path} with {connector.connector_name}'s definition id.")
+    logger.info(f"Updated {definitions_mask_path} with {connector.connector_name}'s definition id.")
     return definitions_mask_path
 
+
 def run_generate_cloud_connector_catalog(airbyte_cloud_repo_path: Path) -> str:
     result = subprocess.check_output(
-        f"cd {airbyte_cloud_repo_path} && ./gradlew :cloud-config:cloud-config-seed:generateCloudConnectorCatalog", 
-        shell=True
-        )
-    logging.info("Ran generateCloudConnectorCatalog Gradle Task")
+        f"cd {airbyte_cloud_repo_path} && ./gradlew :cloud-config:cloud-config-seed:generateCloudConnectorCatalog", shell=True
+    )
+    logger.info("Ran generateCloudConnectorCatalog Gradle Task")
     return result.decode()
 
+
 def commit_all_files(airbyte_cloud_repo: git.Repo, commit_message: str):
-    airbyte_cloud_repo.git.add('--all')
+    airbyte_cloud_repo.git.add("--all")
     airbyte_cloud_repo.git.commit(m=commit_message)
-    logging.info(f"Committed file changes.")
+    logger.info("Committed file changes.")
+
 
-def push_branch(airbyte_cloud_repo: git.Repo, branch:str):
-    airbyte_cloud_repo.git.push("--set-upstream", "origin", branch)
-    logging.info(f"Pushed branch {branch} to origin")
+def push_branch(airbyte_cloud_repo: git.Repo, branch: str):
+    airbyte_cloud_repo.git.push("--force", "--set-upstream", "origin", branch)
+    logger.info(f"Pushed branch {branch} to origin")
 
-def deploy_new_connector_to_cloud_repo(
-    airbyte_cloud_repo_path: Path,
-    airbyte_cloud_repo: git.Repo,
-    connector: ConnectorQAReport
- ):
+
+def pr_already_created_for_branch(head_branch: str) -> bool:
+    response = requests.get(
+        AIRBYTE_PLATFORM_INTERNAL_PR_ENDPOINT,
+        headers=GITHUB_API_COMMON_HEADERS,
+        params={"head": f"{AIRBYTE_PLATFORM_INTERNAL_REPO_OWNER}:{head_branch}", "state": "open"},
+    )
+    response.raise_for_status()
+    return len(response.json()) > 0
+
+
+def create_pr(connector: ConnectorQAReport, branch: str) -> Optional[requests.Response]:
+    body = f"""The Cloud Availability Updater decided that it's the right time to make {connector.connector_name} available on Cloud!
+    Technical name: {connector.connector_technical_name}
+    Version: {connector.connector_version}
+    Definition ID: {connector.connector_definition_id}
+    OSS sync success rate: {connector.sync_success_rate}
+    OSS number of connections: {connector.number_of_connections}
```

Comment:
Nit: Maybe the body should have markdown bullet points?

Reply:
Done